src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -189,20 +239,32 @@ private IServiceCollection BuildCommonServices()
             var services = new ServiceCollection();
             services.AddSingleton(_hostingEnvironment);
 
+            var builder = new ConfigurationBuilder();
```

Comment:
Yeah, it didn't appear that config had a way to do that already. So we need to either add a feature to config or manually do it here. It might be better to do it manually, add the scoped environment variables provider as the first thing then apply any SetSettings that have been used against it.